api/client/logs.go

```diff
@@ -37,8 +37,8 @@ func (cli *DockerCli) CmdLogs(args ...string) error {
 		return err
 	}
 
-	if logType := c.HostConfig.LogConfig.Type; logType != "json-file" {
-		return fmt.Errorf("\"logs\" command is supported only for \"json-file\" logging driver (got: %s)", logType)
+	if logType := c.HostConfig.LogConfig.Type; logType != "json-file" && logType != "journald" {
```

Comment:
Probably best to just remove this check (and no need to pull the container at all as it does above).
The API will error out if the logger is now a supported.
Kinda weird we are doing this extra API call when the API errors anyway.